tests/ui/async-await/in-trait/unconstrained-impl-region.rs

```diff
@@ -14,6 +14,7 @@ impl<'a> Actor for () {
 //~^ ERROR the lifetime parameter `'a` is not constrained by the impl trait, self type, or predicates
     type Message = &'a ();
     async fn on_mount(self, _: impl Inbox<&'a ()>) {}
+    //~^ ERROR the trait bound `impl Inbox<&'a ()>: Inbox<&'a ()>` is not satisfied
```

Comment:
Where does this come from? :thinking:

Reply:
In `compare_impl_item`:
1. the error region is introduced to the hybrid ParamEnv after normalization: `normalize_param_env_or_error(impl Inbox<&'a u8> + ?Sized as Inbox<<() as Actor>::Message>) = impl Inbox<&'a u8> as Inbox<&'ReError u8>)`. Preexisting.
2. When selecting `impl Inbox<&'a u8> as Inbox<&'a u8>`, The param env candidate is ignored because it references errors. New behviour in the PR.